static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -176,6 +192,28 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
     ];
   }
 
+  setPreset(preset: Preset, context: PresetContext) {
+    this.form.setInitialData({
+      ...this.form.initialData,
+      name: context.name,
+      dataset: context.dataset,
+      eventTypes: context.eventTypes as any,
+      aggregate: context.aggregate,
+      comparisonDelta: context.comparisonDelta,
+      timeWindow: context.timeWindow,
+      query: context.query,
```

Comment:
@Neo-Zhixing You can't do `...context`?

Reply:
Can't here, because we're doing `setInitialData` only on a subset of `PresetContext`'s properties. Other properties needs to be set in other ways.